packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -88,14 +93,45 @@ type CreateSearchIndexCancelledAction = {
   type: ActionTypes.CreateSearchIndexCancelled;
 };
 
+type OpenUpdateSearchIndexModalAction = {
+  type: ActionTypes.OpenUpdateSearchIndexModal;
+  indexName: string;
+  indexDefinition: string;
```

Comment:
If we just store the `indexName` in the state and then get the definition when the modal is open (when connecting component with redux)?

Reply:
I've though about it, no strong opinion TBH. I did it this way because in the component that triggers the action [IndexActions](./packages/compass-indexes/src/components/search-indexes-table/search-index-actions.tsx), we already have the latestDefinition and the indexName. It was straightforward to pass them up and we get all the information from the same place so it's pretty obvious where the data comes from.

Reply:
this came up during the Technical Design and we agreed to have it such - just have the index identifier in `update` object and use that to find the corresponding index definition.